$NetBSD: patch-makeinclude.in,v 1.2 2024/07/12 10:27:14 micha Exp $

Do not use the static libraries created in "../lib".
Use libtool.
Upstream will not merge this (libtool is listed as not acceptable).

--- makeinclude.in.orig	2023-12-09 13:58:40.000000000 +0000
+++ makeinclude.in
@@ -100,7 +100,7 @@ LINKFLTKIMG	= @LINKFLTKIMG@ @LINKFLTK@ $
 LINKFLTKCAIRO	= @LINKFLTKCAIRO@ $(CAIROLIBS)
 FLTKCAIROOPTION = @FLTKCAIROOPTION@
 LINKSHARED	= @DSOLINK@ @LINKSHARED@ $(IMAGELIBS) $(CAIROLIBS)
-IMAGELIBS	= -L../lib @IMAGELIBS@
+IMAGELIBS	= @IMAGELIBS@
 
 # image libraries to build...
 IMAGEDIRS	= @JPEG@ @ZLIB@ @PNG@
@@ -152,20 +152,20 @@ UNINSTALL_DESKTOP = @UNINSTALL_DESKTOP@
 
 .o$(EXEEXT):
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(LDFLAGS) $< $(LINKFLTK) $(LDLIBS) -o $@
+	$(LIBTOOL) --mode=link $(CXX) $(ARCHFLAGS) $(LDFLAGS) $< $(LINKFLTK) $(LDLIBS) -o $@
 	$(OSX_ONLY) ../fltk-config --post $@
 
 .c.o:
 	echo Compiling $<...
-	$(CC) -I.. $(ARCHFLAGS) @PNGINC@ @JPEGINC@ @ZLIBINC@ $(CFLAGS) -c $< -o $@
+	$(LIBTOOL) --mode=compile $(CC) -I.. $(ARCHFLAGS) @PNGINC@ @JPEGINC@ @ZLIBINC@ $(CFLAGS) -c $< -o $@
 
 .cxx.o:
 	echo Compiling $<...
-	$(CXX) -I.. $(ARCHFLAGS) @PNGINC@ @JPEGINC@ @ZLIBINC@ $(CXXFLAGS) -c $< -o $@
+	$(LIBTOOL) --mode=compile $(CXX) -I.. $(ARCHFLAGS) @PNGINC@ @JPEGINC@ @ZLIBINC@ $(CXXFLAGS) -c $< -o $@
 
 .mm.o:
 	echo Compiling $<...
-	$(CXX) -I.. $(ARCHFLAGS) $(CXXFLAGS) -x objective-c++ -c $< -o $@
+	$(LIBTOOL) --mode=compile $(CXX) -I.. $(ARCHFLAGS) $(CXXFLAGS) -x objective-c++ -c $< -o $@
 
 
 .man.0 .man.1 .man.3 .man.6:
